src/vsphere_cpi/spec/integration/cluster_different_datastores_spec.rb

```diff
@@ -33,72 +33,123 @@
       'cpu' => 1,
     }
   end
-  let(:first_datastore_cpi) do
-    options = cpi_options(
-      datacenters: [{
-        datastore_pattern: @datastore_pattern,
-        persistent_datastore_pattern: @datastore_pattern,
-      }],
-    )
-    VSphereCloud::Cloud.new(options)
-  end
-  let(:second_datastore_cpi) do
-    options = cpi_options(
-      datacenters: [{
-        datastore_pattern: @second_datastore,
-        persistent_datastore_pattern: @second_datastore,
-      }],
-    )
-    VSphereCloud::Cloud.new(options)
+
+  context 'randomization for datastore selection is set to zero' do
+    # Mocking out stable random to a constant 1 to ensure algorithm scores
+    # datastores in descending order of free space.
+    module VSphereCloud::SelectionPipeline::StableRandom
+      def self.[](object)
+        1
+      end
+    end
+    let(:all_datastore_cpi) do
+      options = cpi_options(
+        datacenters: [{
+          datastore_pattern: '.*',
+          persistent_datastore_pattern: '.*',
+        }],
+      )
+      VSphereCloud::Cloud.new(options)
+    end
+
+    it 'should pick the datastore with most amount of free space' do
```

Comment:
I think this test should be part of a new set of integration tests eg datastore_selection_spec.rb. So, that we can more integration tests related to datastore selection over there and this file will remain a test for testing using different cpis

Reply:
The last heading of file before I added my changes was `given cpis that are configured to use same cluster but different datastores`
That is exactly what I solved. We can instead rename this file to what you suggested.